### PR TITLE
Fix `fs.open` resource leaks

### PR DESCRIPTION
**Pull request by tkukurin**
Invoking `{load,save}_from_dict` results in resource leak warnings, this should fix.
Introduces no significant logic changes.

### CODE REVIEW

**Review by mariosasko — APPROVED** [review]
Made some smaller modifications to improve style and fix the failing test. Thanks!

**Comment by tkukurin** [comment]
@mariosasko Sorry, I didn't check tests/style after doing a merge from the Git UI last week. Thx for fixing. 

FYI I'm getting "Only those with [write access](https://docs.github.com/articles/what-are-the-different-access-permissions) to this repository can merge pull requests." so it seems somebody else needs to merge this.

**Review by lhoestq — APPROVED** [review]
thanks :) merging

**Pull request merged**
